site/lib/src/components/copy_button.dart

```diff
@@ -28,20 +28,32 @@ class CopyButton extends StatefulComponent {
 
 class _CopyButtonState extends State<CopyButton> {
   bool _hidden = true;
+  bool _copied = false;
 
   @override
   void initState() {
-    // Unhide the copy button if successfully initialized on the client.
     if (kIsWeb && component.toCopy.isNotEmpty) {
       _hidden = false;
     }
-
     super.initState();
   }
 
+  void copy() {
+    web.window.navigator.clipboard.writeText(component.toCopy);
+
+    setState(() => _copied = true);
+
+    Future.delayed(const Duration(seconds: 2), () {
```

Comment:
To provide static feedback if we unintentionally return something from the delayed closure:

```suggestion
    Future<void>.delayed(const Duration(seconds: 2), () {
```